pkg/parse/state.go

```diff
@@ -159,14 +159,38 @@ func (s *reconcilerState) resetAllButSourceState() {
 
 // needToSetSourceStatus returns true if `p.setSourceStatus` should be called.
 func (s *reconcilerState) needToSetSourceStatus(newStatus sourceStatus) bool {
-	return !status.HasTransientErrors(newStatus.errs) && (!newStatus.equal(s.sourceStatus) ||
-		s.sourceStatus.lastUpdate.IsZero() ||
-		s.sourceStatus.lastUpdate.Before(&s.syncingConditionLastUpdate))
+	if status.HasTransientErrors(newStatus.errs) {
+		return false
+	}
+	// Update if not initialized
+	if s.sourceStatus.lastUpdate.IsZero() {
+		return true
+	}
+	// Update if source status was last updated before the rendering status
+	if s.sourceStatus.lastUpdate.Before(&s.renderingStatus.lastUpdate) {
```

Comment:
I think the reason that we were comparing with the lastUpdate timestamp before is we thought the sourceStatus and renderingStatus were from the R*Sync status, but it is actually the internal cache from the new commit.
It makes sense to compare with the next phase.